Show custom features in the add modal on top

SARodrigues opened the pull request:
## Show custom features in the add dropdown on top

### Overview

This PR makes it so that, in the _"Add features"_ modal, custom features (user uploaded) are shown at the very top of the list. In order to do so, it makes use of the `isCustom` property added to the respective endpoint in https://github.com/Vizzuality/marxan-cloud/pull/683. 

### Testing instructions

1. Open a project  
2. Click to _"Edit"_ a scenario  
3. Click on the _"Features"_ tab   
4. Click the _"Add features"_ button  
    - Verify that any custom features (if none, upload one) are shown at the very top, before all other features  

### Feature relevant tickets

[MARXAN-967](https://vizzuality.atlassian.net/browse/MARXAN-967)

kgajowy commented on app/layout/scenarios/edit/features/add/list/component.tsx:
```diff
@@ -80,27 +80,29 @@ export const ScenariosFeaturesAddList: React.FC<ScenariosFeaturesAddListProps> =
+              return (
+                <div
+                  key={`${item.id}`}
+                  className={cx({
+                    'border-t border-dashed': i !== 0,
+                  })}
```
[minor / disregard if gibberish]
(not a CSS expert, but maybe it can get some custom, named thing/selector like `:not(:first-child)` ?

SARodrigues replied:
Had to look it up, but you're right. Even though it looks like there's not a `:not` in tailwind, the effect can be replicated fully with Tailwind.

SARodrigues commented on app/hooks/features/index.ts:
```diff
@@ -131,9 +132,12 @@ export function useAllFeatures(projectId, options: UseFeaturesOptionsProps = {})
     })) : [];
 
+    // We want to return custom features first, but preserve the overall sorting
+    const partitionedParsedData = flatten(partition(parsedData, (feature) => feature.isCustom));
```
Not really a fan of the variable name, but couldn't come up with a better name. Hopefully the comment clears any confusion.

kgajowy replied:
`sorted` / `sortedByCustomFeature` seems fine 🤔

SARodrigues commented:
Addressed the suggestions, rebased on `develop`. Also changed the implementation a little as to preserve the sorting as returned by the api endpoint.

kgajowy commented on app/layout/scenarios/edit/features/add/list/component.tsx:
```diff
@@ -80,27 +80,29 @@ export const ScenariosFeaturesAddList: React.FC<ScenariosFeaturesAddListProps> =
+                      handleToggleSelected(item.id);
+                    }}
+                  />
+                </div>
```
@SARodrigues `selected was on purpose - as verbs (`is selected`) rather indicates the function to check something. Take a look at some libraries on even `Item` component - it also uses `selected` ;)

[really minor]
